### ISSUE TITLE
FreeBSD time.h fields timezone & daylight need to tell truth or throw Exception.

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2659]
There are two `time.scala` files in Scala Native. Due to the historical development of Scala Native and 
the `between & betwixt` current separation of `posixlib` and `clib`, the file of interest is in `posixlib`.

FreeBSD (and, I believe, macOS) define two non-Posix fields in a `struct tm`, These fields are
`tm_zone` & `tm_gmtoff`. FreeBSD does not define the the two Posix variables `timezone`
and `daylight`.

As a *convenience*m Scala Native `posixlib/src/main/resources/scala-native/time.c
` provides these fields unconditionally:
```
// XSI                                                                            
  @name("scalanative_timezone")
  def timezone(): CLong = extern

  // XSI                                                                            
  @name("scalanative_tzname")
  def tzname(): Ptr[CStruct2[CString, CString]] = extern

```

`resources/scala-native/time.c`  synthesizes these fields:
```
// For simplicity, socially avoiding Issue #2657, use of () on variable declarations

// XSI                                                                              
long scalanative_timezone() {
#if defined(__FreeBSD__)
    return 0;
#else
    return timezone;
#endif
}

// XSI                                                                              
int scalanative_daylight() {
#if defined(__FreeBSD__)
    return 0;
#else
    return daylight;
#endif
}

```

To the point, in each case, on FreeBSD the 'polite fiction', a.k.a lie, of always
zero is returned.  This makes the variables worse than useless on FreeBSD.
Anyone using them would expect them to either report an error or be
as close to truth as possible.

If one goes the `correct` but less than useful path of reporting attempted
uses on FreeBSD, then the report should come from the .scala code, as
an Exception, and not from C code as a signal. Scala Native does not, 
and probably should not, convert C signals to Scala Native exceptions.
C signals tend to cry havoc in Scala Native programs.

Providing an accurate result for `timezone`, my current concern,
probably means (in C) allocating a `struct tm`, and calling something
like `localtime()` with `time(NULL) a.k.a now`. One can then return
the tm_gmtoffset` field.

Providing `daylight` requires more thought. The `is_dst` field
is known, by some, to be unreliable. Implementing `daylight`
is left as an exercise for the gentle reader.

Of course, all of this presupposes having an machine
running FreeBSD for testing. This is the poster child
for special case code.

**Comment by LeeTibbert** [comment]
Caveat:

The fix for `timezone` suggested in the base note may/probably_is
too simple.

One should explore on an actual FreeBSD system if `tm_gmtoff`
returns the standard or daylight adjusted time if `is_dst` is true.
Perhaps `is_dst` is accurate.

On may need the algorithm of getting timeNow, calling
localtime() and gmtime() and figuring out the difference.
Perhaps only doing this  only if `is_dst` is true?

All this is by saying that experimentation & experience on
an actual FreeBSD system is needed before hacking.
Otherwise, we will all be at a pain point again. I remember
those and they were no fun.

**Comment by LeeTibbert** [comment]
Throwing an Exception was probably the better way to go, but trying to implement that
looked just to hairy.  I went with synthesizing the values on FreeBSD.

There can be a __slight__ difference in behavior between the FreeBSD and Linux, etc
variants when `tzset()` has not been previously called.  It depends on if daylight
savings is in effect.  When daylight savings are in effect, the FreeBSD variant
can act as though `tzset()` had been called. Not having called `tzset()` previously
is usually a bug.

I mention this possible difference in case someone gets motivated to write an
acceptance test.

**Comment by LeeTibbert** [comment]
Fixed in merged PR #2695
